pkgs/top-level/all-packages.nix

```diff
@@ -2032,7 +2032,7 @@ with pkgs;
 
   xpaste = callPackage ../tools/text/xpaste { };
 
-  xrootd = callPackage ../tools/networking/xrootd {
+  xrootd = callPackage ../by-name/xr/xrootd/package.nix {
     # Workaround systemd static build breakage
     systemd = if systemd.meta.broken then null else systemd;
```

Comment:
We should make this dependend on whether we build static and then don't include systemd instead of overwriting it with null.

Reply:
Using a `lib.optionalAttrs` for instance ?

Reply:
Actually, couldn't we simply have this logic within the derivation instead of putting it in `all-packages.nix` ?

Reply:
I'm not sure whether it was useful or not... I am trying to completely remove this.

Reply:
> I'm not sure whether it was useful or not... I am trying to completely remove this.

@ShamrockLee what do you think about this ?

Reply:
This was originally an attempt to statically build CERN ROOT (`root`).

> Actually, couldn't we simply have this logic within the derivation instead of putting it in `all-packages.nix` ?

It can be moved into `package.nix` by optionally injecting systemd into `buildInputs`.

Reply:
Update: I cannot make `pkgsStatic.xrootd` build so far, as `pkgsStatic.scitokens-cpp` fails to build. Let's move on and fix it later.

Reply:
> It can be moved into `package.nix` by optionally injecting systemd into `buildInputs`.

Currently, I add it systematically to `buildInputs` (as soon as the hostPlatform is linux).
Do you want me to introduce back a conditional on `systemd.meta.broken` ?

Like so for instance:
```
-    ++ lib.optionals stdenv.hostPlatform.isLinux [
-      systemd
-      voms
-    ];
+    ++ lib.optionals stdenv.hostPlatform.isLinux (
+      [
+        voms
+      ]
+      ++ lib.optional (!systemd.meta.broken) systemd
+    );
```

Reply:
The `systemd` package now vetoes static host platforms using `badPlatform` instead of `broken`.

The following works

```patch
-    ++ lib.optionals stdenv.hostPlatform.isLinux [
-      systemd
-      voms
-    ];
+    ++ lib.filter (lib.meta.availableOn stdenv.hostPlatform) [
+      [
+        systemd # only available on specific non-static Linux platforms
+        voms # only available on Linux due to gsoap failing to build on Darwin
+      ]
+      ++ lib.optional (!systemd.meta.broken) systemd
+    );
```

Reply:
Done, I updated it.